[CS2113-W14-3] FinanceBuddy

limkongkiat opened the pull request:
Finance Buddy is a software that allows university students to log their daily expenditures to help manage their budgets. Users can add, delete and edit expenditure logs into the app as well as list out all their logged transactions.

thomasjlalba commented on src/main/java/seedu/duke/command/AddIncomeCommand.java:
```diff
@@ -0,0 +1,54 @@
+        int preEntryCount = list.getEntryCount();
+        list.addEntry(income);
+        assert list.getEntryCount() == preEntryCount + 1 : "Income not added";
+        System.out.println("--------------------------------------------");
+        System.out.println("Got it! I've added this income:");
+        System.out.println(income);
+        System.out.println("--------------------------------------------");
```
might be good to refactor this such that it is not magic strings

thomasjlalba commented on src/main/java/seedu/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,41 @@
+    public void execute(FinancialList list) {
+        if (index > 0 && index <= list.getEntryCount()) {
+            assert index > 0 && index <= list.getEntryCount();
+            FinancialEntry entry = list.getEntry(index - 1);
+            list.deleteEntry(index - 1);  // Index correction as list is 0-based
```
Maybe you can refactor `index - 1` as a variable since it is used multiple times, and also to make it clearer that it is because list is 0-based

thomasjlalba commented on src/main/java/seedu/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    @Override
+    public void execute(FinancialList list) {
+        if (index > 0 && index <= list.getEntryCount()) {
+            assert index > 0 && index <= list.getEntryCount();
```
Since it is used multiple times, you could refactor the condition statement as a boolean variable

thomasjlalba commented on src/main/java/seedu/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,41 @@
+            System.out.println("Okay! The following entry has been deleted: ");
+            System.out.println(entry);
+            System.out.println("--------------------------------------------");
+        } else {
+            System.out.println("OOPS!!! The entry does not exist.");
```
You can make the [happy path more prominent](https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) by making this a guard clause instead

thomasjlalba commented on src/main/java/seedu/duke/command/SeeAllIncomesCommand.java:
```diff
@@ -0,0 +1,62 @@
+ * Command to print all incomes recorded in the financial list.
+ */
+public class SeeAllIncomesCommand extends Command {
+    private LocalDate start;
+    private LocalDate end;
```
The variable name might be a bit ambiguous, maybe it could be better refined as `startDate`?

thomasjlalba commented on src/main/java/seedu/duke/command/SeeAllIncomesCommand.java:
```diff
@@ -0,0 +1,62 @@
+     * @return true if entry should be listed out, false otherwise.
+     */
+    private boolean shouldBeIncluded(FinancialEntry entry) {
+        return entry instanceof Income && (end == null || entry.getDate().isBefore(end))
+                && (start == null || entry.getDate().isAfter(start));
```
Maybe you could avoid such [complicated expressions](https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions) and look to refactor it to make it simpler to read

thomasjlalba commented on src/main/java/seedu/duke/command/SeeAllIncomesCommand.java:
```diff
@@ -0,0 +1,62 @@
+        for (int i = 0; i < list.getEntryCount(); i++) {
+            FinancialEntry entry = list.getEntry(i);
+            if (shouldBeIncluded(entry)) {
+                incomeList += ((++incomeCount) + ". " + entry + System.lineSeparator());
```
Maybe you can bring out the increment to make the code more readable and obvious

thomasjlalba commented on src/main/java/seedu/duke/ui/AppUi.java:
```diff
@@ -0,0 +1,287 @@
+            throw new FinanceBuddyException("Please enter a valid start/end date");
+        }
+
+        LocalDate startDate = start != null ? DateParser.parse(commandArguments.get("/from")) : null;
+        LocalDate endDate = end != null ? DateParser.parse(commandArguments.get("/to")) : null;
```
Maybe you can look to make the expression more readable, or some comments to explain the process

thomasjlalba commented on src/main/java/seedu/duke/ui/AppUi.java:
```diff
@@ -0,0 +1,287 @@
+        LocalDate startDate = start != null ? DateParser.parse(commandArguments.get("/from")) : null;
+        LocalDate endDate = end != null ? DateParser.parse(commandArguments.get("/to")) : null;
+
+        if (type != null) {
```
Maybe you can use a guard clause instead to prevent deep nesting?

thomasjlalba replied:
Maybe you can look into extracting this out as a method on its own?

thomasjlalba commented on src/main/java/seedu/duke/financial/FinancialList.java:
```diff
@@ -0,0 +1,75 @@
+    public FinancialEntry getEntry(int index) {
+        if (index >= 0 && index <= entries.size()) {
+            return entries.get(index);
+        } else {
+            System.out.println("OOPS!!! The entry does not exist.");
+            return null;
+        }
```
Maybe this can be converted into an error?

thomasjlalba commented on src/main/java/seedu/duke/ui/AppUi.java:
```diff
@@ -0,0 +1,287 @@
+     *                         and the amount ("/a") and the date/time ("/dt")
+     */
+    public void addExpense(HashMap<String, String> commandArguments) {
+        String description = commandArguments.get("argument");
```
Might be good to refactor these magic strings

thomasjlalba commented on src/main/java/seedu/duke/ui/AppUi.java:
```diff
@@ -0,0 +1,287 @@
+     */
+    public void listHelper(HashMap<String, String> commandArguments) throws FinanceBuddyException {
+        String type = commandArguments.get("argument");
+        String start = commandArguments.get("/from");
+        String end = commandArguments.get("/to");
```
Maybe you can better name these variables to `startDateString`, as it might be confusing alongside `startDate` and `endDate`.

thomasjlalba commented on src/test/java/seedu/duke/command/AddExpenseCommandTest.java:
```diff
@@ -0,0 +1,162 @@
+        addExpenseCommand.execute(financialList);
+
+        String output = outputStream.toString();
+        String expectedOutput = "--------------------------------------------" + System.lineSeparator() +
```
Maybe can keep the consistency between `\n` and `System.lineSeperator()`

thomasjlalba commented on src/main/java/seedu/duke/parser/InputParser.java:
```diff
@@ -0,0 +1,59 @@
+     * @param input The raw input string from the user.
+     * @return A HashMap containing the parsed command and its arguments.
+     */
+    public static HashMap<String, String> parseCommands(String input) {
```
Maybe you can look to splitting this method to multiple simpler methods

thomasjlalba reviewed (commented):
Some areas can be simplified to make the logic and flow more obvious and simpler to read. Keep up the good work!

hzxnancy commented on docs/UML/SeeAllEntriesOverview.png:
Generally a well drawn diagram, but would be better if there are activation bars!

telferang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,537 @@
+The diagram below shows the inheritance of the `Command` class. The diagram is only meant to show
+the hierarchy of classes and have been greatly simplified.
+
+![Command_Inheritance_Diagram](UML/CommandInheritance.png)
```
Consider using a normal rectangle instead of a rounded rectangle

telferang replied:
Consider switching the direction of the arrow

telferang replied:
Consider using an empty arrow instead of a filled arrow

ZiliaAJY commented on docs/UML/CommandInheritance.png:
I found quite a few rounded rectangles, consider using normal rectangle boxes.

ZiliaAJY commented on docs/UML/SeeAllEntriesExecution.png:
Objects are missing semi-colons, for example :Logic (in quite a few sequence diagrams).

ZiliaAJY commented on docs/UML/SeeAllEntriesOverview.png:
The parameters are missing the corresponding types.

ZiliaAJY commented on docs/UML/SeeAllEntriesOverview.png:
Certain conditions in 'alt' can be indicated as "else".

ZiliaAJY reviewed (commented):
Overall good job!

telferang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,537 @@
+The interaction between the command classes and the `FinancialList` is as follows,
+using `SeeAllEntriesCommand` as an example:
+
+![execution](UML/SeeAllEntriesExecution.png)
```
Consider adding [else] for the alt frame

telferang replied:
Consider adding a ":" before an entities

hzxnancy commented on docs/UML/SeeAllEntriesExecution.puml:
```diff
@@ -0,0 +1,44 @@
+actor user
+participant "SeeAllEntriesCommand" as cmd
+participant "FinancialList" as list
+participant "FinancialEntry" as entry
```
Would it be better to start the entity's name with a colon?

telferang reviewed (commented):
The diagram is mostly correct. I added some comments for your reference.

hzxnancy commented on docs/UML/overallFlow.png:
May I ask if this is an activity diagram? If it is, it would be better to use coloured cirlces to denote the start and end of the program! 
![image](https://github.com/user-attachments/assets/f16965b7-bb45-4b47-9d58-eb4d7d4eb39f)

hzxnancy reviewed (commented):
Overall, its a really good DP, as each command have a lot of details and the DP is structured neatly! Just take note of the minor details and should be alright.

Toby-Yu commented on docs/UML/SeeAllEntriesExecution.png:
The sequence diagram seems miss the activation bar.

Toby-Yu commented on docs/UML/overallFlow.png:
This structural diagram is clear.

Toby-Yu commented on .gitignore:
Is good to put data into the gitignore

Toby-Yu commented on docs/DeveloperGuide.md:
There is no example of input and output screenshots for the explanation with the DG.

Toby-Yu reviewed (commented):
Overall DG is ok but there is something needs to be added

tsoudibi committed:
Fix checkStyle

JuvenLimHY committed:
Update DateParser

JuvenLimHY committed:
Edit UG

JuvenLimHY committed:
Add no / in description

JuvenLimHY committed:
Update date format dd/MM/yyyy

JuvenLimHY committed:
Update year to yyyy

JuvenLimHY committed:
Update PPP

JuvenLimHY committed:
Merge pull request #280 from Littleme1032/Littleme1032-Juven-edit-ppp

Update PPP

ctzeyong committed:
Merge branch 'master' into Budget

ctzeyong committed:
Update budget prompting to user on startup

limkongkiat committed:
Add line breaks in Help sample output

limkongkiat committed:
Merge pull request #275 from limkongkiat/update-UG-post-ped

Update UG post ped

ctzeyong committed:
Update when balance recalculation is done

ctzeyong committed:
Prevent budget amount from exceeding 9 999 999

ctzeyong committed:
Minor refactoring of method name to more appropriately suit what it does

ctzeyong committed:
Update JUnit tests for added functionality

ctzeyong committed:
Update JavaDoc for new methods and JUnit tests

tsoudibi committed:
Merge pull request #277 from tsoudibi/add-storage-feedback

Add-storage-feedback

tsoudibi committed:
Update edit UG

tsoudibi committed:
Seperate SD into 2

JuvenLimHY committed:
Merge branch 'master' into Littleme1032-juven-add-Date-Time-Feature

JuvenLimHY committed:
Edit test case

limkongkiat committed:
Merge pull request #276 from limkongkiat/bugfix-edit-out-of-position

Bugfix edit out of position

JuvenLimHY committed:
Edit UG

JuvenLimHY committed:
Merge pull request #279 from Littleme1032/Littleme1032-juven-add-Date-Time-Feature

Update DateParser

JuvenLimHY committed:
Merge pull request #278 from Littleme1032/Littleme1032-Juven-edit-UG

Edit UG

limkongkiat committed:
Change list command to show global index instead of relative index

list command previously showed entries with index relative only to the
entries listed. This does not really help the user.

Let's:
* Change SeeAllEntriesCommand to display the index of each entry in
the FinancialList so that the user immediately knows which index to
edit / delete should he wish to do so.
* Display the total count of entries listed so that the user still
immediately knows at a glance how many entries they have listed.

limkongkiat committed:
Change displayed indexes and add total count for tests for `list`

ctzeyong committed:
Saves budget and balance amount as 2dp

ctzeyong committed:
Ensure budget and balance amounts are displayed the correct number of times

ctzeyong committed:
Update JUnit tests

ctzeyong committed:
Update JavaDoc

ctzeyong committed:
Minor checkstyle edit

tsoudibi committed:
Merge branch 'master' into update-ug-dg

tsoudibi committed:
Merge pull request #282 from tsoudibi/update-ug-dg

Update edit UG

limkongkiat committed:
Update Javadoc comment for execute method of SeeAllEntriesCommand

limkongkiat committed:
Add test for SeeAllEntriesCommand to check handling of null list

limkongkiat committed:
Extract repeated code in list command tests

limkongkiat committed:
Update test in Logic for handling list command

limkongkiat committed:
Reduce magic strings in list command tests

limkongkiat committed:
Extract repeated code in SeeAllEntriesCommandTest into new method

limkongkiat committed:
Reduce magic strings in SeeAllEntriesCommandTest

limkongkiat committed:
Add Javadoc comments for helper function in SeeAllExpensesCommandTest

kestryix committed:
Add JUnit test for input parser

JuvenLimHY committed:
Add DeleteAll Feature

JuvenLimHY committed:
Edit checkstyle

limkongkiat committed:
Merge pull request #283 from limkongkiat/edit-show-global-index

Edit List command to show list of entries in financial list instead of relative index

kestryix committed:
Modify budget command to take in argument

kestryix committed:
Merge branch 'master' into app-budget

kestryix committed:
Fix checkstyle

kestryix committed:
Fix checkstyle

kestryix committed:
Fix checkstyle

kestryix committed:
Fix checkstyle

limkongkiat committed:
Merge pull request #281 from ctzeyong/Budget

BudgetLogic Updates

limkongkiat committed:
Update sequence diagrams for List command

limkongkiat committed:
Update examples for list command in UG

limkongkiat committed:
Update PPP for Lim Kong Kiat

kestryix committed:
Merge branch 'master' into app-budget

limkongkiat committed:
Update description for list command in UG

kestryix committed:
Fix checkstyle

kestryix committed:
Fix checkstyle

kestryix committed:
Update boolean name

kestryix committed:
Merge pull request #285 from kestryix/app-budget

Modify budget command to take in argument

tsoudibi committed:
Add function to delete  Files

limkongkiat committed:
Add check to limit financial list to 5000 entries maximum

limkongkiat committed:
Add test for AddExpenseCommand to test 5000-entry limit

limkongkiat committed:
Add test for AddIncomeCommand to test 5000 entry limit

limkongkiat committed:
Add error signature to SeeAllExpensesCommandTest method

limkongkiat committed:
Add test to ensure edit entry works with full list of 5000 entries

tsoudibi committed:
Fix junit

limkongkiat committed:
Make minor edits to pass checkstyle

tsoudibi committed:
Merge pull request #288 from tsoudibi/add-delete-budget-storage

Add function to delete files

kestryix committed:
Update reposense link

limkongkiat committed:
Add back Storage files

limkongkiat committed:
Merge pull request #287 from limkongkiat/update-UG-DG-list-entries

Update List Entries section in UG & DG

kestryix committed:
Update storage when budget deleted

tsoudibi committed:
Update output when no budget file

tsoudibi committed:
Fix checkStyle

tsoudibi committed:
Fix typo

tsoudibi committed:
Fix junit

tsoudibi committed:
Merge pull request #291 from tsoudibi/fix-storage-output

Update output when no budget file

kestryix committed:
Merge branch 'refs/heads/master' into app-budget

kestryix committed:
Update storage when budget deleted

kestryix committed:
Update PPP

kestryix committed:
Merge pull request #293 from kestryix/verity-ppp

Update PPP

JuvenLimHY committed:
Add function to delete range

limkongkiat committed:
Merge pull request #289 from limkongkiat/limit-max-entries

Limit Maximum number of entries

JuvenLimHY committed:
Revert back indextoAmend

accidentally removed the feature of last entries delete

ctzeyong committed:
Merge branch 'master' into PPP

kestryix committed:
Fix bug

limkongkiat committed:
Merge pull request #284 from Littleme1032/Littleme1032-juven-DeleteAll

Add DeleteAll Feature

kestryix committed:
Merge pull request #292 from kestryix/app-budget

Update storage when budget deleted

kestryix committed:
Fix error message

ctzeyong committed:
Update PPP

ctzeyong committed:
Merge branch 'master' into Budget

tsoudibi committed:
Modify storage warning

tsoudibi committed:
Merge pull request #301 from tsoudibi/edit-storage-file-warning

Modify storage warning

kestryix committed:
Fix checkstyle

kestryix committed:
Fix testcase

kestryix committed:
Merge pull request #302 from kestryix/app-budget

Fix invalid budget amount error message

ctzeyong committed:
Update logic to prompt user to set budget

ctzeyong committed:
Merge remote-tracking branch 'upstream/master' into Budget

ctzeyong committed:
Update logic on whether to print balance amount

limkongkiat committed:
Update Help Command and corresponding test

ctzeyong committed:
Update balance recalculation order when setting budget

limkongkiat committed:
Edit budget command description in Help command

ctzeyong committed:
Edit logic for prompting user to set budget to prevent null
Also edit JUnit tests

ctzeyong committed:
Update JavaDoc

ctzeyong committed:
Merge pull request #304 from ctzeyong/Budget

Budget Bug Fixing

tsoudibi committed:
Update msg

tsoudibi committed:
Update Junit

tsoudibi committed:
Merge pull request #305 from tsoudibi/update-storage-msg

Update-storage-msg

limkongkiat committed:
Add check to ensure that entry is not unchanged after edit command

limkongkiat committed:
Extract error message to Commons class

limkongkiat committed:
Change BUDGET to AMOUNT in Help message

tsoudibi committed:
update ppp

limkongkiat committed:
Merge pull request #303 from limkongkiat/update-help-command-111124

Update Help Command and corresponding test

limkongkiat committed:
Merge pull request #307 from limkongkiat/edit-command-check-if-changed

Edit command check if changed

tsoudibi committed:
Update content

tsoudibi committed:
Merge pull request #308 from tsoudibi/tony-update-ppp

Update ppp

ctzeyong committed:
PPP update

ctzeyong committed:
Merge pull request #309 from ctzeyong/PPP

Update PPP

kestryix committed:
Update budget and command summary sections

ctzeyong committed:
Merge branch 'master' into UserGuide

ctzeyong committed:
Update UG

kestryix committed:
Merge pull request #310 from kestryix/verity-ug

Update budget and command summary sections

ctzeyong committed:
Merge pull request #311 from ctzeyong/UserGuide

User Guide Update

kestryix committed:
Update logic sequence diagram

limkongkiat committed:
Fix bugs related to delete command
Fix out of bounds index error message
Extract magic strings to Commons class

limkongkiat committed:
Fix checkstyle errors

tsoudibi committed:
Update Storage UG and DG

limkongkiat committed:
Merge pull request #312 from limkongkiat/fix-delete-bugs

Fix bugs related to delete command

ctzeyong committed:
Update DG

ctzeyong committed:
Merge branch 'master' into DeveloperGuide

tsoudibi committed:
Add more use cases in DG

tsoudibi committed:
Update comment of storage

tsoudibi committed:
Add another use case

kestryix committed:
Update ui, parser and logic sections based on current code

limkongkiat committed:
Extract constants and rounding method into Commons class

limkongkiat committed:
Add Javadoc comments for Commons class

kestryix committed:
Merge pull request #314 from kestryix/verity-dg

Update UI, Parser and Logic sections

kestryix committed:
Add duplicate argument detection to input parser

limkongkiat committed:
Merge pull request #315 from limkongkiat/update-commons

Update commons

kestryix committed:
Merge branch 'master' into app-budget

kestryix committed:
Fix checkstyle

kestryix committed:
Merge pull request #316 from kestryix/app-budget

Add duplicate argument detection to input parser

kestryix committed:
Update PPP format

ctzeyong committed:
DG update

ctzeyong committed:
Merge branch 'master' into Budget

kestryix committed:
Merge pull request #317 from kestryix/verity-ppp

Update PPP format

ctzeyong committed:
Update prompting when budget was set in a previous month

ctzeyong committed:
Update JUnit test

ctzeyong committed:
Merge pull request #319 from ctzeyong/DeveloperGuide

Developer Guide Update

ctzeyong committed:
Merge pull request #318 from ctzeyong/Budget

Update to user prompt when budget was set in a previous month

tsoudibi committed:
Update Edit UG

tsoudibi committed:
Update comtent

tsoudibi committed:
Fix typo

tsoudibi committed:
Add newlines

tsoudibi committed:
Merge pull request #313 from tsoudibi/tony-update-ug-dg-v3

Update Storage UG and DG

ctzeyong committed:
DG update

ctzeyong committed:
Merge pull request #320 from ctzeyong/master

DG update

tsoudibi committed:
update typo

ctzeyong committed:
Merge branch 'master' into PPP

tsoudibi committed:
Merge pull request #321 from tsoudibi/tony-update-ug-again

update typo

ctzeyong committed:
Update PPP

limkongkiat committed:
Update Command Summary of UG

limkongkiat committed:
Update List Entries section of UG

limkongkiat committed:
Update Headers for Set/Edit Budget examples in UG

limkongkiat committed:
Update Overview and Implementation Section of DG

limkongkiat committed:
Update User Stories and Product Scope

limkongkiat committed:
Add Future Enhancements Section to DG

limkongkiat committed:
Add Future Enhancements to Table of Contents in DG

limkongkiat committed:
Fix typo in PPP

limkongkiat committed:
Update description in Overview section of DG

JuvenLimHY committed:
Update Delete

limkongkiat committed:
Merge pull request #322 from limkongkiat/update-UG-DG-PPP-kk

Update UG/DG/PPP KK

JuvenLimHY committed:
Update DG

JuvenLimHY committed:
Update UserGuide.md

ctzeyong committed:
Merge branch 'master' into DeveloperGuide

ctzeyong committed:
DG update

limkongkiat committed:
Update PPP

limkongkiat committed:
Format User Guide sections

limkongkiat committed:
Update diagram in DG

JuvenLimHY committed:
Merge pull request #323 from Littleme1032/Littleme1032-juven-UG-DG

Edit Delete component in  Ug and DG

limkongkiat committed:
Reorganize User Stories in DG

ctzeyong committed:
UML sequence diagram update

ctzeyong committed:
Logging update for DG

ctzeyong committed:
Merge pull request #324 from ctzeyong/DeveloperGuide

Developer Guide

ctzeyong committed:
Merge branch 'master' into PPP

limkongkiat committed:
Update PPP

ctzeyong committed:
Update PPP

ctzeyong committed:
Merge pull request #325 from ctzeyong/PPP

PPP

limkongkiat committed:
Merge pull request #326 from limkongkiat/update-docs

Update docs

kestryix committed:
Update Ui Parser Diagram

kestryix committed:
Merge pull request #327 from kestryix/verity-dg

Update Ui Parser Diagram

kestryix committed:
Update PPP

kestryix committed:
Merge pull request #328 from kestryix/verity-ppp

Update PPP

JuvenLimHY committed:
Removed duplicate

JuvenLimHY committed:
Update UserGuide.md

JuvenLimHY committed:
changed all to dd/MM/yyyy

JuvenLimHY committed:
Update example

kestryix committed:
Update about us

kestryix committed:
Merge pull request #330 from kestryix/verity-ppp

Update about us

kestryix committed:
Update PPP format

ctzeyong committed:
Merge branch 'master' into DeveloperGuide

kestryix committed:
Merge pull request #331 from kestryix/verity-ppp

Update PPP format

ctzeyong committed:
Change Finance Buddy to FinanceBuddy

tsoudibi committed:
ADD last amended transaction to DU and fix UG typo

tsoudibi committed:
Merge pull request #332 from tsoudibi/fix-ug-typo

ADD last amended transaction to DU and fix UG typo

ctzeyong committed:
Minor formatting fix

JuvenLimHY committed:
Cleaning up Notes

JuvenLimHY committed:
Update line

ctzeyong committed:
Edit DG

ctzeyong committed:
Merge pull request #333 from ctzeyong/DeveloperGuide

Change Finance Buddy to FinanceBuddy in DG

JuvenLimHY committed:
Merge pull request #329 from Littleme1032/UG-duplicate

Removed duplicate

ctzeyong committed:
Merge branch 'master' into PPP

ctzeyong committed:
Update PPP

ctzeyong committed:
Merge pull request #334 from ctzeyong/PPP

PPP Update

limkongkiat committed:
Update DG

ctzeyong committed:
Update PPP

limkongkiat committed:
Update DG

limkongkiat committed:
Update Diagrams

ctzeyong committed:
Merge pull request #335 from ctzeyong/PPP

Update PPP

ctzeyong committed:
Merge branch 'master' into DeveloperGuide

JuvenLimHY committed:
Update add transaction

ctzeyong committed:
Update DG UML

JuvenLimHY committed:
update

limkongkiat committed:
Add Page Break to DG

ctzeyong committed:
Merge pull request #337 from ctzeyong/DeveloperGuide

Developer Guide

limkongkiat committed:
Remove extra colon in Contents Page

limkongkiat committed:
Merge pull request #338 from limkongkiat/update-DG-final

Update DG final

limkongkiat committed:
Merge pull request #336 from Littleme1032/UG-duplicate

Ug duplicate

JuvenLimHY committed:
update

JuvenLimHY committed:
Merge pull request #339 from Littleme1032/UG-duplicate

update

JuvenLimHY committed:
Update UserGuide.md

JuvenLimHY committed:
Merge pull request #340 from Littleme1032/UG-duplicate

Update UserGuide.md

JuvenLimHY committed:
update

JuvenLimHY committed:
Merge pull request #341 from Littleme1032/Ug-update

update

JuvenLimHY committed:
update

JuvenLimHY committed:
Merge pull request #342 from Littleme1032/Ug-update

update

limkongkiat committed:
Update PPP

ctzeyong committed:
Merge branch 'master' into DeveloperGuide

limkongkiat committed:
Merge pull request #343 from limkongkiat/final-update-ppp

Update PPP

ctzeyong committed:
Fix broken links

ctzeyong committed:
Merge pull request #344 from ctzeyong/DeveloperGuide

Developer guide

ctzeyong committed:
Merge branch 'master' into UserGuide

ctzeyong committed:
Minor bug fix

ctzeyong committed:
Merge pull request #345 from ctzeyong/UserGuide

User guide

limkongkiat committed:
Update UG

limkongkiat committed:
Merge pull request #346 from limkongkiat/last-update-ug

Update UG
